pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -0,0 +1,256 @@
+    distance between `U` and `V`, the value of the HST is 0 if and only if `V` is equal to `U` (up to global phase).
+    Therefore we can define a cost by:
+
+    .. math::
+        C_{HST} = 1 - \frac{1}{d^2} \left|Tr(V^{\dagger}U)\right|^2,
+
+    where the quantity :math:`\frac{1}{d^2} \left|Tr(V^{\dagger}U)\right|^2` is the Hilbert Schmidt Test. It is
+    equivalent to taking the outcome probability of the state :math:`|0 ... 0\rangle` for the following circuit:
+
+    .. figure:: ../../_static/templates/subroutines/hst.png
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    It defines our decomposition for the Hilbert Schmidt Test template.
+
+    Args:
+        params (array): Parameters for the quantum function `V`.
+        v_function (Callable): Quantum function that represents the approximate compiled unitary `V`.
+        v_wires (int or Iterable[Number, str]]): the wire(s) the approximate compiled unitary act on.
+        u_tape (.QuantumTape): `U`, the unitary to be compiled as a ``qml.tape.QuantumTape``.
+
+    Raises:
+        QuantumFunctionError: The argument u_tape must be a QuantumTape.
+        QuantumFunctionError: ``v_function`` is not a valid Quantum function.
+        QuantumFunctionError: `U` and `V` do not have the same number of wires.
+        QuantumFunctionError: The wires ``v_wires`` are a subset of `V` wires.
+        QuantumFunctionError: u_tape and v_tape must act on distinct wires.
+
+    .. UsageDetails::
+
+        Consider that we want to evaluate the Hilbert Schmidt Test cost between the unitary `U` and an approximate
+        unitary `V`. We need to define some functions where it is possible to use the ``.HilbertSchmidt`` template.
+        Here the considered unitary is ``Hadamard`` and we try to compute the cost for the approximate unitary ``RZ``.
+        For an angle which is equal to 0 (identity), we have the maximal cost which is 1.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import HilbertSchmidt
+
+            with qml.tape.QuantumTape(do_queue=False) as U:
+                qml.Hadamard(wires=0)
+
+            def v_circuit(params, v_wires):
+                qml.RZ(params[0], wires=v_wires[0])
+
+            dev = qml.device("default.qubit", wires=4)
+
+            @qml.qnode(dev)
+            def hilbert_test(v_params, v_function, v_wires, u_tape):
+                qml.HilbertSchmidt(v_params, v_function=v_function, v_wires=v_wires, u_tape=u_tape)
+                return qml.probs(u_tape.wires + v_wires)
+
+            def cost_hst(parameters, v_function, v_wires, u_tape):
+                return (1 - hilbert_test(v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape)[0])
+
+        Now that you have defined your cost function you can call it for specific parameters.
+
+        >>> cost_hst([0], v_function = v_circuit, v_wires = [1], u_tape = U)
```

Comment:
This line may not be up to the formatting standards of `black`.

Reply:
I don't understand why it would not work here?

Reply:
Yes it is for computing the decomposition.

Reply:
Docstrings are not being formatted by `black`. The comment is just a note that the formatting might be off for the example. We're not being super tedious with formatting examples in the docstrings, but it can be nicer for users to have the examples conform to the formatting produced by `black`.